### PR TITLE
Stack alloc

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

File: gtk/TreeStore.custom [hunk 1]
```diff
@@ -310,56 +310,60 @@
 
 		private TreeIter InsertWithValues (bool hasParent, TreeIter parent, int position, params object[] values)
 		{
-			int[] columns = new int[values.Length];
-			GLib.Value[] vals = new GLib.Value[values.Length];
-			int n_values = 0;
+			unsafe {
```
Comment (Therzok) [inline gtk/TreeStore.custom]: This is the most important optimization, as we invoke this pretty often in MD.

**Comment by Therzok** [comment]
https://github.com/mono/gtk-sharp/pull/207/commits/60c3220fe47acdaeefbda8e5e4aa77cda3701dca

Means:
Faster allocation - not on the heap
No more GValueArray wrapper on the heap - plus, no more finalization for it.

**Comment by Therzok** [comment]
https://github.com/mono/gtk-sharp/pull/207/commits/60c3220fe47acdaeefbda8e5e4aa77cda3701dca
https://github.com/mono/gtk-sharp/pull/207/commits/199e4614e19d1dba6b40edf63a6a41d17a212a36

Result in this diff:
https://gist.github.com/Therzok/af03aa73dfde604a9f57c0e14db0d589

**Comment by alanmcgovern** [comment]
Also one thing to keep in mind - there are no guarantees about the contents of stackalloced memory. It looks like we always fully initialize all memory we allocate this way, but if you ever want allocate an extra byte for a null terminator (or whatever) do remember to explicitly write null to that byte!

**Comment by Therzok** [comment]
The stackallocs are mostly GLib.Value/Utf8String via marshal, so it's safe to assume we don't get uninitialized memory.

**Pull request merged**